addons/mail/models/mail_activity.py

```diff
@@ -837,7 +839,7 @@ def _search_activity_state(self, operator, value):
     @api.depends('activity_ids.date_deadline')
     def _compute_activity_date_deadline(self):
         for record in self:
-            record.activity_date_deadline = record.activity_ids[:1].date_deadline
+            record.activity_date_deadline = record.my_activity_date_deadline = record.activity_ids[:1].date_deadline
```

Comment:
This is not correct: you should get the deadline of the first activity of the current user.
```suggestion
            record.activity_date_deadline = record.activity_ids[:1].date_deadline
            record.my_activity_date_deadline = next((
                activity.date_deadline
                for activity in record.activity_ids
                if activity.user_id.id == self.env.uid
            ), False)
```
Also, `my_activity_date_deadline` depends on the current user, so you need another compute method decorated with `@api.depends_context('uid')`.